roles/rsyslog/tasks/main_core.yml

```diff
@@ -364,7 +364,6 @@
       service:
         name: rsyslog
         enabled: true
-        state: started
```

Comment:
On latest Fedora this causes error messages in /var/log/messages as such:
```
May 25 09:43:31 7e33d79e-6b6e-4e22-97fc-fa84e46170bf rsyslogd: error during parsing file /etc/rsyslog.d/00-global.conf, on or before line 12: parameter 'workdirectory' specified more than once - one instance is ignored. Fix config [v8.2312.0-1.fc40 try https://www.rsyslog.com/e/2207 ]
May 25 09:43:31 7e33d79e-6b6e-4e22-97fc-fa84e46170bf rsyslogd: error during parsing file /etc/rsyslog.d/00-global.conf, on or before line 12: parameter 'workDirectory' not known -- typo in config file? [v8.2312.0-1.fc40 try https://www.rsyslog.com/e/2207 ]
May 25 09:43:31 7e33d79e-6b6e-4e22-97fc-fa84e46170bf rsyslogd: parameters for built-in module builtin:omfile already set - ignored  [v8.2312.0-1.fc40 try https://www.rsyslog.com/e/2220 ]
May 25 09:43:31 7e33d79e-6b6e-4e22-97fc-fa84e46170bf rsyslogd: module 'imuxsock' already in this config, cannot be added  [v8.2312.0-1.fc40 try https://www.rsyslog.com/e/2221 ]
May 25 09:43:31 7e33d79e-6b6e-4e22-97fc-fa84e46170bf rsyslogd: error during parsing file /etc/rsyslog.d/90-input-basics-basic_input.conf, on or before line 8: parameter 'SysSock.Use' not known -- typo in config file? [v8.2312.0-1.fc40 try https://www.rsyslog.com/e/2207 ]
May 25 09:43:31 7e33d79e-6b6e-4e22-97fc-fa84e46170bf rsyslogd: error during parsing file /etc/rsyslog.d/90-input-basics-basic_input.conf, on or before line 8: parameter 'SysSock.RateLimit.Interval' not known -- typo in config file? [v8.2312.0-1.fc40 try https://www.rsyslog.com/e/2207 ]
```

This is because the main config file was not generated yet. Why this does not fail on RHEL 8 and 9 - I have no idea.
But it seems like a good practice to start a service using a handler after the role invocation finishes.

Reply:
> On latest Fedora this causes error messages in /var/log/messages as such

Only on Fedora 40, but not 39, or on other platforms?  Sounds like an issue with the fedora 40 rsyslog package.

> This is because the main config file was not generated yet

The rsyslog package provides a default /etc/rsyslog.conf (and possibly files in /etc/rsyslog.d/), and the service should start with the default files.  If this isn't working, it is a problem with the rsyslog package.

> But it seems like a good practice to start a service using a handler after the role invocation finishes.

In general, yes, but in some cases the role needs the service to be running before it can take further actions.